File: 2525.根据规则将箱子分类.go
/*
 * @lc app=leetcode.cn id=2525 lang=golang
 *
 * [2525] 根据规则将箱子分类
 */

// @lc code=start
func categorizeBox(length int, width int, height int, mass int) string {
	heavy, bulky := false, false
	if length >= 1e4 || width >= 1e4 || height >= 1e4 || (length*width*height) >= 1e9 {
		bulky = true
	}
	if mass >= 100 {
		heavy = true
	}
	if heavy && bulky {
		return "Both"
	}
	if heavy {
		return "Heavy"
	}
	if bulky {
		return "Bulky"
	}
	return "Neither"

}

// @lc code=end

File: 2525.根据规则将箱子分类.py
#
# @lc app=leetcode.cn id=2525 lang=python3
#
# [2525] 根据规则将箱子分类
#

# @lc code=start
class Solution:
    def categorizeBox(self, length: int, width: int, height: int, mass: int) -> str:
        Heavy = False
        Bulky = False
        if length >= 10000 or width >= 10000 or height >= 10000 or (length * width * height) >= 10**9:
            Bulky = True
        if mass >= 100:
            Heavy = True
        if Heavy and Bulky:
            return "Both"
        if Heavy:
            return "Heavy"
        if Bulky:
            return "Bulky"
        return "Neither"
# @lc code=end

